### PR TITLE
[embedded] Mark all functions as 'nounwind' in embedded Swift, add dependency tests

### PR DESCRIPTION
**Pull request by kubamracek**
Currently, some (not all) IRGen'd functions end up emiting unwind info and referencing exception personality symbols, both of which is undesirable in embedded Swift. Let's mark all functions an NoUnwind and let's add a test that checks that a simple embedded Swift program in Mach-O and ELF files has exactly the expected dependencies.

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by rauhul** [comment]
Nice! I can't wait to drop these hacks!
```
@_cdecl("__aeabi_unwind_cpp_pr0")
func __unused_0() { fatalError() }

@_cdecl("__aeabi_unwind_cpp_pr1")
func __unused_1() { fatalError() }
```

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by eeckstein** [comment]
Maybe a stupid question: what about C++ interop? Do we restrict C++ interop to not supporting exceptions with this?

**Comment by kubamracek** [comment]
Changed the condition to only apply 'nounwind' if C++ interop is disabled.

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by hyp** [comment]
exceptions can be caught when C++ interop is enabled.

File: lib/IRGen/IRGenModule.cpp [hunk 1]
```diff
@@ -1444,6 +1444,11 @@ void IRGenModule::constructInitialFnAttributes(
     Attrs.addAttribute(llvm::Attribute::StackProtectReq);
     Attrs.addAttribute("stack-protector-buffer-size", llvm::utostr(8));
   }
+
+  if (Context.LangOpts.hasFeature(Feature::Embedded) &&
+      !Context.LangOpts.EnableCXXInterop) {
+    Attrs.addAttribute(llvm::Attribute::NoUnwind);
```
Comment (hyp) [inline lib/IRGen/IRGenModule.cpp:1450]: Is that applied only to Swift functions, and not the imported C++ functions? In this case you can enable `nounwind` for Swift functions, as exceptions are always caught at the boundary between Swift and a call to C++.

Comment (kubamracek) [reply lib/IRGen/IRGenModule.cpp:1450]: I think you're right, this only applies to SILFunctions, not imported C++ functions. @eeckstein is that correct?

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test Linux platform

**Comment by kubamracek** [comment]
@swift-ci please test Linux platform

**Pull request merged**